test/jdk/javax/swing/JFileChooser/NewFolderLocale/FileChooserNewFolderLocaleTest.java

```diff
@@ -0,0 +1,116 @@
+        if (IS_WINDOWS) {
+            newFolderKey = "FileChooser.win32.newFolder";
+            newFolderSubKey = "FileChooser.win32.newFolder.subsequent";
+        } else {
+            newFolderKey = "FileChooser.other.newFolder";
+            newFolderSubKey = "FileChooser.other.newFolder.subsequent";
+        }
+
+        try {
+            JFileChooser fileChooser = new JFileChooser();
+            File currentDir = new File(".");
+            newFolderEnglish =
+                    fileChooser.getFileSystemView().createNewFolder(currentDir);
+
+            setLocale("fr");
+            fileChooser = new JFileChooser();
+            newFolderFrench =
+                    fileChooser.getFileSystemView().createNewFolder(currentDir);
+            if( !newFolderFrench.getName().contains
+                    (UIManager.getString(newFolderKey))) {
+                throw new RuntimeException("Failed to update new Locale");
+            }
+        } finally {
+            if (newFolderEnglish != null && newFolderEnglish.delete()) {
+                System.out.println("Deleted the file: " +
+                        newFolderEnglish.getName());
+            } else {
+                System.out.println("Failed to delete file : " +
+                        newFolderEnglish.getName());
+            }
+            if (newFolderFrench != null && newFolderFrench.delete()) {
+                System.out.println("Deleted the file: " +
+                        newFolderFrench.getName());
+            } else {
+                System.out.println("Failed to delete file : " +
+                        newFolderFrench.getName());
+            }
+        }
+        System.out.println("Test Passed!");
+    }
+
+    private static void setLocale(String locale) {
+        res = ResourceBundle.getBundle("bundle",
+                new Locale(locale));
+        UIManager.put("FileChooser.acceptAllFileFilterText",
+                res.getString("accept_all"));
+        UIManager.put("FileChooser.directoryOpenButtonText",
+                res.getString("open_text"));
+        UIManager.put("FileChooser.openButtonText",
+                res.getString("open_text"));
+        UIManager.put("FileChooser.saveButtonText",
+                res.getString("save_text"));
+        UIManager.put("FileChooser.cancelButtonText",
+                res.getString("cancel_text"));
+        UIManager.put("FileChooser.lookInLabelText",
+                res.getString("file_look"));
+        UIManager.put("FileChooser.saveInLabelText",
+                res.getString("file_save"));
+        UIManager.put("FileChooser.fileNameLabelText",
+                res.getString("file_name"));
+        UIManager.put("FileChooser.filesOfTypeLabelText",
+                res.getString("file_type"));
+        UIManager.put(newFolderKey,
+                res.getString("new_folder"));
+        UIManager.put(newFolderSubKey,
+                res.getString("new_folder") + " ({0})");
```

Comment:
The only thing needed here is:
```java
        UIManager.put(newFolderKey, "Nouveau dossier");
        UIManager.put(newFolderSubKey, "Nouveau dossier ({0})");
```
And this could be inlined into the test method code.